src/www/app/app.ts

```diff
@@ -153,61 +153,79 @@ export class App {
       messageKey = 'error-feedback-submission';
-    } else if (e instanceof errors.ServerUrlInvalid) {
+    } else if (error instanceof errors.ServerUrlInvalid) {
       messageKey = 'error-invalid-access-key';
-    } else if (e instanceof errors.ServerIncompatible) {
+    } else if (error instanceof errors.ServerIncompatible) {
       messageKey = 'error-server-incompatible';
-    } else if (e instanceof OperationTimedOut) {
+    } else if (error instanceof OperationTimedOut) {
       messageKey = 'error-timeout';
-    } else if (e instanceof errors.ShadowsocksStartFailure && this.isWindows()) {
+    } else if (error instanceof errors.ShadowsocksStartFailure && this.isWindows()) {
       // Fall through to `error-unexpected` for other platforms.
       messageKey = 'outline-plugin-error-antivirus';
       buttonKey = 'fix-this';
       buttonLink = 'https://s3.amazonaws.com/outline-vpn/index.html#/en/support/antivirusBlock';
-    } else if (e instanceof errors.ConfigureSystemProxyFailure) {
+    } else if (error instanceof errors.ConfigureSystemProxyFailure) {
       messageKey = 'outline-plugin-error-routing-tables';
       buttonKey = 'feedback-page-title';
       buttonHandler = () => {
         // TODO: Drop-down has no selected item, why not?
         this.rootEl.changePage('feedback');
       };
-    } else if (e instanceof errors.NoAdminPermissions) {
+    } else if (error instanceof errors.NoAdminPermissions) {
       messageKey = 'outline-plugin-error-admin-permissions';
-    } else if (e instanceof errors.UnsupportedRoutingTable) {
+    } else if (error instanceof errors.UnsupportedRoutingTable) {
       messageKey = 'outline-plugin-error-unsupported-routing-table';
-    } else if (e instanceof errors.ServerAlreadyAdded) {
+    } else if (error instanceof errors.ServerAlreadyAdded) {
       messageKey = 'error-server-already-added';
-      messageParams = ['serverName', this.getServerDisplayName(e.server)];
-    } else if (e instanceof errors.SystemConfigurationException) {
+      messageParams = ['serverName', this.getServerDisplayName(error.server)];
+    } else if (error instanceof errors.SystemConfigurationException) {
       messageKey = 'outline-plugin-error-system-configuration';
-    } else if (e instanceof errors.ShadowsocksUnsupportedCipher) {
+    } else if (error instanceof errors.ShadowsocksUnsupportedCipher) {
       messageKey = 'error-shadowsocks-unsupported-cipher';
-      messageParams = ['cipher', e.cipher];
+      messageParams = ['cipher', error.cipher];
+    } else if (error instanceof errors.ServerAccessKeyInvalid) {
+      messageKey = 'error-connection-configuration';
+      buttonKey = 'error-details';
+      buttonHandler = () => {
+        this.showErrorDetailDialog(error);
+      };
+    } else if (error instanceof errors.SessionConfigFetchFailed) {
+      messageKey = 'error-connection-configuration-fetch';
+      buttonKey = 'error-details';
```

Comment:
I think it would probably be cleaner and easier to understand if we were localizing here:
```suggestion
      buttonLabel = this.localize('error-details');
```

Later you can just check if it's undefined.